### ISSUE TITLE
`jj init` only checks PWD for existing repos

### ISSUE THREAD

**Issue by tingerrr** [martinvonz/jj#4173]
## Description
I was in a subdirectory of a git repo and noticed it was not yet a colocated jj repo, so I ran `jj git init --colocate` and it created both a `.git` and `.jj` directory in the PWD. This is not quite what I expected. I wasn't aware of the optional positional `DESTINATION` parameter.

As it turns out, jj will happily create a new repo of any kind, even if there already is a repo in a parent directory.

## Steps to Reproduce the Problem
1. Create a repository with git or jj
1. Create and navigate to a subdirectory
1. Create a repository using `jj init`

## Expected Behavior
jj looks up the parent directories additionally to the PWD and behaves otherwise the same, i.e. it fails for already existing repos and suggests using `--colocate` for git repos. When doing `jj git init --colocate` within the subdirectory of a git repository, it does the conversion at the right git repo root. Passing a `DESTINAITON` can be used to force the nested repository, but destinations like `.` or another subdirectory of the parent repository should at least cause a warning.

Alternatively, it could simply look up and warn the user about the likely unintended nested repository instead of throwing an error.

## Actual Behavior
jj will create a new nested repository without any warning or error.

## Specifications
- Platform: `Linux twinkbook 6.10.2-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 27 Jul 2024 16:49:55 +0000 x86_64 GNU/Linux`
- Version: `0.19.0`

**Comment by yuja** [comment]
Nested Git repository is perfectly valid, so `jj git init --colocate` in Git sub tree directory should be allowed. That said, it might make sense to warn if `--colocate`, and if there's `.git` directory in parent, and if no `.jj` directory exists there.

**Comment by tingerrr** [comment]
> Nested Git repository is perfectly valid

I suppose it is, but what's the use case for this. It seems, to me at least, that something like this is only rarely intended. Which is why I suggested making the seemingly more common use case (colocation/conversion of parent repo) the default without making the explicit sub repo creation impossible (`jj git init .` with warning).

Although now that I think about it, if a user indeed intended to create a sub repository, then reversing the conversion/colocation of the parent repo would be hard.

Perhaps this would make more sense, if we had a command to uncolocate/convert a repo back to it's backing storage type.

**Comment by yuja** [comment]
Maybe we shouldn't overload `jj git init --colocate` as a command to either create or import Git repo? It would be confusing if `init` created a repo in different directory depending on the existence of a `.git` directory in parent.

> I suppose it is, but what's the use case for this.

Someone might create `.git` in their home/project directory? (I wouldn't recommend such setup, but that should be doable.)

**Comment by tingerrr** [comment]
Yeah, that's probably true, the current behavior is easy to reverse if one knows it happened, so the warning is all that's missing in my opinion.

**Comment by lf-** [comment]
I would anticipate `jj git init --colocate` to make a repo in `git rev-parse --show-toplevel` as well. The current behaviour should possibly require a `--force`, though `git init` *also* does this silently and yet I have never once done mistakes with it; this is probably a sign that `jj git init` has different typical usage than `git init` :).
